vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/vaadin-grid-flow-selection-column.js

```diff
@@ -1,156 +1,97 @@
-      ];
-    }
-
-    /** @private */
-    connectedCallback() {
-      super.connectedCallback();
-      if (this._grid) {
-        this._grid.addEventListener('select', this._boundOnSelectEvent);
-        this._grid.addEventListener('deselect', this._boundOnDeselectEvent);
-      }
-    }
-
-    /** @private */
-    disconnectedCallback() {
-      super.disconnectedCallback();
-      if (this._grid) {
-        this._grid.removeEventListener('select', this._boundOnSelectEvent);
-        this._grid.removeEventListener('deselect', this._boundOnDeselectEvent);
-      }
-    }
-
-    /**
-     * Renders the Select All checkbox to the header cell.
-     *
-     * @override
-     */
-    _defaultHeaderRenderer(root, _column) {
-      let checkbox = root.firstElementChild;
-      if (!checkbox) {
-        checkbox = document.createElement('vaadin-checkbox');
-        checkbox.id = 'selectAllCheckbox';
-        checkbox.setAttribute('aria-label', 'Select All');
-        checkbox.classList.add('vaadin-grid-select-all-checkbox');
-        checkbox.addEventListener('click', this._onSelectAllClick.bind(this));
-        root.appendChild(checkbox);
-      }
+class GridFlowSelectionColumn extends GridSelectionColumnBaseMixin(GridColumn) {
 
-      const checked = this.selectAll;
-      checkbox.hidden = this.selectAllHidden;
-      checkbox.checked = checked;
-      checkbox.indeterminate = this.indeterminate;
-    }
+  static get is() {
+    return 'vaadin-grid-flow-selection-column';
+  }
 
-    /**
-     * Renders the Select Row checkbox to the body cell.
-     *
-     * @override
-     */
-    _defaultRenderer(root, _column, { item, selected }) {
-      let checkbox = root.firstElementChild;
-      if (!checkbox) {
-        checkbox = document.createElement('vaadin-checkbox');
-        checkbox.setAttribute('aria-label', 'Select Row');
-        checkbox.addEventListener('click', this._onSelectClick.bind(this));
-        root.appendChild(checkbox);
+  static get properties() {
+    return {
+      /**
+       * Override property to enable auto-width
+       */
+      autoWidth: {
+        type: Boolean,
+        value: true
+      },
+
+      /**
+       * Override property to set custom width
+       */
+      width: {
+        type: String,
+        value: '56px'
       }
+    };
+  }
 
-      checkbox.__item = item;
-      checkbox.checked = selected;
-    }
-
-    _onSelectClick(e) {
-      e.currentTarget.checked ? this._grid.$connector.doDeselection([e.currentTarget.__item], true) : this._grid.$connector.doSelection([e.currentTarget.__item], true);
+  /**
+   * Override method from `GridSelectionColumnBaseMixin` to add ID to select all
+   * checkbox
+   *
+   * @override
+   */
+  _defaultHeaderRenderer(root, _column) {
+    super._defaultHeaderRenderer(root, _column);
+    const checkbox = root.firstElementChild;
+    if (checkbox) {
+      checkbox.id = 'selectAllCheckbox';
     }
+  }
```

Comment:
This should be cleaned up as well, right now every grid defines this ID, which means there will be elements with duplicate IDs in the DOM.

Could use a class name that is set by the header renderer as alternative. However there is no TestBench method for getting the checkbox, so this would also be a breaking change as developers are probably relying on the ID as well.